validator/client/src/main/java/tech/pegasys/teku/validator/client/duties/BlockProductionDuty.java

```diff
@@ -101,47 +103,17 @@ public SafeFuture<BLSSignature> createRandaoReveal(final ForkInfo forkInfo) {
     return validator.getSigner().createRandaoReveal(spec.computeEpochAtSlot(slot), forkInfo);
   }
 
-  // TODO: sign blob sidecars
-  public SafeFuture<SignedBeaconBlock> signBlock(
-      final ForkInfo forkInfo, final Optional<BlockContainer> maybeBlock) {
-    final BeaconBlock unsignedBlock =
-        maybeBlock
-            .map(BlockContainer::getBlock)
-            .orElseThrow(
-                () -> new IllegalStateException("Node was not syncing but could not create block"));
+  public SafeFuture<SignedBlockContainer> signBlockContainer(
+      final ForkInfo forkInfo, final Optional<BlockContainer> maybeBlockContainer) {
+    final BlockContainer unsignedBlockContainer =
+        maybeBlockContainer.orElseThrow(
+            () -> new IllegalStateException("Node was not syncing but could not create block"));
```

Comment:
What are the relations with syncing? There are a lot of focus on it in the message.
Maybe describe what we really have: Block container from BeaconNode is empty. BeaconNode provided incorrect unsigned block container. Something around it

Reply:
if the node is syncing it return an HTTP status with error and will be handled differently. In this case you received an empty response. I think we can leave it, has been there since beacon launch :)